### ISSUE TITLE
Expand templating to include WCAG/508 criteria

### ISSUE THREAD

**Issue by grugnog** [GSA/open-product-accessibility-template#14]

**Comment by dmundra** [comment]
@mgifford comparing the current output to https://github.com/CivicActions/accessibility/blob/main/VPAT/Drupal9AccessibilityConformanceReport.md is there sections we are missing that should be added? We would likely tackle them in the next sprint but maybe we can start by documenting it here. What do you think?

**Comment by mgifford** [comment]
We're pretty close now.  At least for WCAG.  There are other 508 criteria we need to add, I'll follow up with that in another comment.

Whether we include the author as different than the vendor is another issue, but I think something like this would be useful:

## Contacts

```
contact:
  vendor: 
    contact_name: 
    organization_name: 
    address: 
    url: 
    email: 
    phone: 
  author:# if different than vendor
    contact_name: 
    organization_name:
    address: 
    email: 
    url: 
    phone: 
```

I've added URL because well, this is digital after all. Might as well. 

## Evaluation Methods Used

Can we just add this:

`  evaluation_methods_used: Use of automated tools like WAVE and Accessibility Insights. Manual keyboard-only testing. Some testing with JAWS, NVDA, and VoiceOver. The evaluation process also includes a review of the Drupal Core accessibility issue queue.`

## Applicable Standards/Guidelines

We are going to define this here for the Librarian, right? So should we be specifying this:

```
applicable_standard_guideline: 
  success_criteria_level_a
  success_criteria_level_aa
  section_508_refresh_2018
```

## Legal Disclaimer 

I don't think we'd need more than one slot for the legal disclaimer. I assume this would be written by the author, but we can leave that. Still need a slot for it though.

```
  legal_disclaimer: The information herein is provided in good faith based on the analysis of the web application at the time of the review and does not represent a legally binding claim. Please contact us to report any accessibility errors or conformance claim errors for re-evaluation and correction, if necessary.
```

**Comment by mgifford** [comment]
To comply with Section 508 we also need to add Chapters 3, 4, 5 & 6.  I think we should try to auto-generate Chapter 3, although this is more programming work, so we may want to just add it in as text for now. 

```
  functional_performance_criteria:
    - num: 302.1
      name: Without Vision
      level: "supports"
    - num: 302.2
      name: With Limited Vision
      level: "supports"
    - num: 302.3
      name: Without Perception of Color
      level: "supports"
    - num: 302.4
      name: Without Hearing
      level: "supports"
    - num: 302.5
      name: With Limited Hearing
      level: "supports"
    - num: 302.6
      name: Without Speech
      level: "supports"
    - num: 302.7
      name: With Limited Manipulation
      level: "supports"
    - num: 302.8
      name: With Limited Reach and Strength
      level: "supports"
    - num: 302.9
      name: With Limited Language, Cognitive, and Learning Abilities
      level: "supports"
```
      

The rest would be mostly empty for our Drupal example, but might include Chapter 4:

```
  hardware:
    - num: 402.2.1
      name: Information Displayed On-Screen
      level: "not-applicable"
    - num: 402.2.2
      name: Transactional Outputs
      level: "not-applicable"
    - num: 402.2.3
      name: Speech Delivery Type and Coordination
      level: "not-applicable"
    - num: 402.2.4
      name: User Control
      level: "not-applicable"
    - num: 402.2.5
      name: Braille Instructions
      level: "not-applicable"
    - num: 402.3.1
      name: Private Listening
      level: "not-applicable"
    - num: 402.3.2
      name: Non-private Listening
      level: "not-applicable"
    - num: 402.4
      name: Characters on Display Screens
      level: "not-applicable"
    - num: 402.5
      name: Characters on Variable Message Signs
      level: "not-applicable"
    - num: 403.1
      name: General (Biometrics)
      level: "not-applicable"
    - num: 404.1
      name: General (Preservation of Information Provided for Accessibility)
      level: "not-applicable"
    - num: 405.1
      name: General (Privacy)
      level: "not-applicable"
    - num: 406.1
      name: General (Standard Connections)
      level: "not-applicable"
    - num: 407.2
      name: Contrast (Operable Parts)
      level: "not-applicable"
    - num: 407.3.1
      name: Tactilely Discernible
      level: "not-applicable"
    - num: 407.3.2
      name: Alphabetic Keys
      level: "not-applicable"
    - num: 407.3.3
      name: Numeric Keys
      level: "not-applicable"
    - num: 407.4
      name: Key Repeat
      level: "not-applicable"
    - num: 407.5
      name: Timed Response
      level: "not-applicable"
    - num: 407.6
      name: Operation
      level: "not-applicable"
    - num: 407.7
      name: Tickets, Fare Cards, and Keycards
      level: "not-applicable"
    - num: 407.8.1
      name: Vertical Reference Plane
      level: "not-applicable"
    - num: 407.8.1.1
      name: Vertical Plane for Side Reach
      level: "not-applicable"
    - num: 407.8.1.2
      name: Vertical Plane for Forward Reach
      level: "not-applicable"
    - num: 407.8.2
      name: Side Reach
      level: "not-applicable"
    - num: 407.8.2.1
      name: Unobstructed Side Reach
      level: "not-applicable"
    - num: 407.8.2.2
      name: Obstructed Side Reach
      level: "not-applicable"
    - num: 407.8.3
      name: Forward Reach
      level: "not-applicable"
    - num: 407.8.3.1
      name: Unobstructed Forward Reach
      level: "not-applicable"
    - num: 407.8.3.2
      name: Obstructed Forward Reach
      level: "not-applicable"
    - num: 407.8.3.2.1
      name: Operable Part Height for ICT with Obstructed Forward Reach
      level: "not-applicable"
    - num: 407.8.3.2.2
      name: Knee and Toe Space under ICT with Obstructed Forward Reach
      level: "not-applicable"
    - num: 408.2
      name: Visibility
      level: "not-applicable"
    - num: 408.3
      name: Flashing
      level: "not-applicable"
    - num: 409.1
      name: General (Status Indicators)
      level: "not-applicable"
    - num: 410.1 
      name: General (Color Coding)
      level: "not-applicable"
    - num: 411.1
      name: General (Audible Signals)
      level: "not-applicable"
    - num: 412.2.1
      name: Volume Gain for Wireline Telephones
      level: "not-applicable"
    - num: 412.2.2
      name: Volume Gain for Non-Wireline ICT
      level: "not-applicable"
    - num: 412.3.1
      name: Wireless Handsets
      level: "not-applicable"
    - num: 412.3.2
      name: Wireline Handsets
      level: "not-applicable"
    - num: 412.4
      name: Digital Encoding of Speech
      level: "not-applicable"
    - num: 412.5
      name: Real-Time Text Functionality
      level: "reserved-for-future"
    - num: 412.6
      name: Caller ID
      level: "not-applicable"
    - num: 412.7
      name: Video Communication
      level: "not-applicable"
    - num: 412.8.1
      name: TTY Connectability
      level: "not-applicable"
    - num: 412.8.2
      name: Voice and Hearing Carry Over
      level: "not-applicable"
    - num: 412.8.3
      name: Signal Compatibility
      level: "not-applicable"
    - num: 412.8.4
      name: Voice Mail and Other Messaging Systems
      level: "not-applicable"
    - num: 413.1.1
      name: Decoding and Display of Closed Captions
      level: "not-applicable"
    - num: 413.1.2
      name: Pass-Through of Closed Caption Data
      level: "not-applicable"
    - num: 414.1.1
      name: Digital Television Tuners
      level: "not-applicable"
    - num: 414.1.2
      name: Other ICT
      level: "not-applicable"
    - num: 415.1.1 
      name: Caption Controls
      level: "not-applicable"
    - num: 415.1.2
      name: Audio Description Controls
      level: "not-applicable"
```

Chapter 5:

```
  software:
    - num: 502.2.1
      name: User Control of Accessibility Features
      level: "not-applicable"
    - num: 502.2.2
      name: No Disruption of Accessibility Features
      level: "not-applicable"
    - num: 502.3.1
      name: Object Information
      level: "not-applicable"
    - num: 502.3.2
      name: Modification of Object Information
      level: "not-applicable"
    - num: 502.3.3
      name: Row, Column, and Headers
      level: "not-applicable"
    - num: 502.3.4
      name: Values
      level: "not-applicable"
    - num: 502.3.5
      name: Modification of Values
      level: "not-applicable"
    - num: 502.3.6
      name: Label Relationships
      level: "not-applicable"
    - num: 502.3.7
      name: Hierarchical Relationships
      level: "not-applicable"
    - num: 502.3.8
      name: Text
      level: "not-applicable"
    - num: 502.3.9
      name: Modification of Text
      level: "not-applicable"
    - num: 502.3.10 
      name: List of Actions
      level: "not-applicable"
    - num: 502.3.11 
      name: Actions on Objects
      level: "not-applicable"
    - num: 502.3.12
      name: Focus Cursor
      level: "not-applicable"
    - num: 502.3.13
      name: Modification of Focus Cursor
      level: "not-applicable"
    - num: 502.3.14
      name: Event Notification
      level: "not-applicable"
    - num: 502.4
      name:  Platform Accessibility Features
      level: "not-applicable"
    - num: 503.2
      name: User Preferences
      level: "not-applicable"
    - num: 503.3
      name: Alternative User Interfaces
      level: "not-applicable"
    - num: 503.4.1
      name:  Caption Controls
      level: "not-applicable"
    - num: 503.4.2
      name: Audio Description Controls
      level: "not-applicable"
    - num: 504.2.1
      name: Preservation of Information Provided for Accessibility in Format Conversion
      level: "not-applicable"
    - num: 504.2.2
      name: PDF Export
      level: "not-applicable"
    - num: 504.3
      name: Prompts
      level: "not-applicable"
    - num: 504.4
      name: Templates
      level: "not-applicable"
```

Chapter 6:

```
  support_documentation_and_services:
    - num: 602.2
      name: Accessibility and Compatibility Features
      level: "not-applicable"
    - num: 602.4
      name: Alternate Formats for Non-Electronic Support Documentation
      level: "not-applicable"
    - num: 603.2
      name: Information on Accessibility and Compatibility Features
      level: "not-applicable"
    - num: 603.3
      name: Accommodation of Communication Needs
      level: "not-applicable"
```

This would address the full Section 508 criteria.

**Comment by dmundra** [comment]
@mgifford I created a separate issue ☝🏾 to tackle 'Applicable Standards/Guidelines' which I will also use to fix the table/chapter output issues. So I think the last item remaining on this issue is adding to the catalog the 508 items.

**Comment by mgifford** [comment]
I think those should all be in here now https://github.com/GSA/open-product-accessibility-template/pull/97

**Comment by dmundra** [comment]
Thanks, @mgifford merged it. I will leave it to you to close this issue.

**Comment by mgifford** [comment]
Thanks!
